pulp_rpm/app/models.py

```diff
@@ -1134,6 +1147,21 @@ def new_version(self, base_version=None):
     class Meta:
         default_related_name = "%(app_label)s_%(model_name)s"
 
+    def finalize_new_version(self, new_version):
+        """
+        Ensure there are no duplicates in a repo version and content is not broken.
+
+        Remove duplicates based on repo_key_fields.
+        TODO: Ensure that modulemd is added with all its RPMs.
```

Comment:
should we open a ticket recursive add/remove of modules or we have one already?

Reply:
we have 2 tickets, one for each, they are in the blocker list